src/mono/mono/metadata/class.c

```diff
@@ -870,6 +870,21 @@ inflate_generic_type (MonoImage *image, MonoType *type, MonoGenericContext *cont
 		nt->data.type = inflated;
 		return nt;
 	}
+	case MONO_TYPE_FNPTR: {
+		MonoMethodSignature *in_sig = type->data.method;
+		// quick bail out - if there are no type variables anywhere in the signature,
+		// there's nothing that could get inflated.
+		if (!in_sig->has_type_parameters)
+			return NULL;
+		MonoMethodSignature *new_sig = mono_inflate_generic_signature (in_sig, context, error);
+		if (!new_sig || !is_ok (error))
+			return NULL;
+		if (new_sig == in_sig)
```

Comment:
can this happen? i'm not opposed to it but it seems weird

Reply:
Yes.
1. The signature inflation code has a cache, so repeated values are possible, in general.
2. The incoming context could put in the same type variables that are already in the sig, so the result is the same as the input.